### PR TITLE
Update PR tester

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

**Comment by karianna** [comment]
```
TestCompilation > compile_pr_test_pipelineTest() STANDARD_ERROR
    org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
    prTester/pr_test_pipeline.groovy: 48: [Static type checking] - No such property: GIT_URL for class: java.lang.Object
     @ line 48, column 71.
       M', userRemoteConfigs: [[url: config.GIT
                                     ^

    prTester/pr_test_pipeline.groovy: 50: [Static type checking] - No such property: disableJob for class: java.lang.Object
     @ line 50, column 34.
               println "${javaVersion} ${config.disableJob}"
                                        ^

    prTester/pr_test_pipeline.groovy: 58: [Static type checking] - No such property: WORKSPACE for class: testDoubles.ContextStub
     @ line 58, column 27.
               println "loading ${context.WORKSPACE}/pipelines/build/common/config_regeneration.groovy"
                                 ^

    prTester/pr_test_pipeline.groovy: 59: [Static type checking] - No such property: WORKSPACE for class: testDoubles.ContextStub
     @ line 59, column 53.
       rationScript = context.load "${context.W
                                     ^

    prTester/pr_test_pipeline.groovy: 59: [Static type checking] - Cannot assign value of type testDoubles.ContextStub$Configuration to variable of type groovy.lang.Closure
     @ line 59, column 51.
       nerationScript = context.load "${context
                                     ^

    prTester/pr_test_pipeline.groovy: 66: [Static type checking] - No such property: WORKSPACE for class: testDoubles.ContextStub
     @ line 66, column 31.
                   println "loading ${context.WORKSPACE}/pipelines/jobs/configurations/jdk${javaVersion}_pipeline_config.groovy"
                                     ^

    prTester/pr_test_pipeline.groovy: 67: [Static type checking] - No such property: WORKSPACE for class: testDoubles.ContextStub
     @ line 67, column 54.
       nfigurations = context.load "${context.W
                                     ^

    prTester/pr_test_pipeline.groovy: 69: [Static type checking] - Cannot find matching method java.lang.Object#regenerate(). Please check if the declared type is correct and if the method exists.
     @ line 69, column 13.
                   regenerationScript(
                   ^
```

File: pipelines/build/prTester/pr_test_pipeline.groovy [hunk 1]
```diff
@@ -17,53 +14,77 @@ See the License for the specific language governing permissions and
...
+                targetConfigurations: testConfigurations
+        ]
+    }
+
+    def generatePipelineJob(def javaVersion) {
+        def config = generateConfig(javaVersion)
+        context.checkout([$class: 'GitSCM', userRemoteConfigs: [[url: config.GIT_URL]], branches: [[name: branch]]])
+
+        println "${javaVersion} ${config.disableJob}"
+        def create = context.jobDsl targets: "pipelines/jobs/pipeline_job_template.groovy", ignoreExisting: false, additionalParameters: config
```
Comment (karianna) [inline pipelines/build/prTester/pr_test_pipeline.groovy:51]: where does create get used?

File: pipelines/build/prTester/pr_test_pipeline.groovy [hunk 2]
```diff
@@ -17,53 +14,77 @@ See the License for the specific language governing permissions and
...
+        javaVersions.each({ javaVersion ->
+            // generate top level job
+            generatePipelineJob(javaVersion)
+            println "[INFO] Running regeneration script..."
+
+            println "loading ${context.WORKSPACE}/pipelines/jobs/configurations/jdk${javaVersion}_pipeline_config.groovy"
+            def buildConfigurations = context.load "${context.WORKSPACE}/pipelines/jobs/configurations/jdk${javaVersion}_pipeline_config.groovy"
+
+            regenerationScript(
+                    "jdk${javaVersion}u",
```
Comment (karianna) [inline pipelines/build/prTester/pr_test_pipeline.groovy:70]: Does tip (`jdk` work here?)

File: pipelines/build/prTester/pr_test_pipeline.groovy [hunk 3]
```diff
@@ -80,22 +102,17 @@ Map<String, ?> defaultTestConfigurations = [
         ]
 ]
 
-// We've emptied this array on purpose until we can get this to work again with
-// the asynchronous nature of our builds
-// TODO https://github.com/AdoptOpenJDK/openjdk-build/issues/1770
-List<Integer> defaultJavaVersions = []
-//List<Integer> defaultJavaVersions = [8, 11, 14, 15]
+List<Integer> defaultJavaVersions = [8, 11]
```
Comment (karianna) [inline pipelines/build/prTester/pr_test_pipeline.groovy:105]: Do we want to add 14 and tip (15) to this?

Comment (johnoliver) [reply pipelines/build/prTester/pr_test_pipeline.groovy:105]: not sure if I can get it working for tip

File: pipelines/build/prTester/pr_test_pipeline.groovy (hunk 2 above)
Comment (johnoliver) [reply pipelines/build/prTester/pr_test_pipeline.groovy:70]: no

File: pipelines/build/prTester/pr_test_pipeline.groovy (hunk 1 above)
Comment (johnoliver) [reply pipelines/build/prTester/pr_test_pipeline.groovy:51]: removed

File: pipelines/build/prTester/pr_test_pipeline.groovy (hunk 3 above)
Comment (karianna) [reply pipelines/build/prTester/pr_test_pipeline.groovy:105]: OK, let's open a separate issue for that.  I'm thinking its useful as most build toolchain changes are now landing on the tip.

**Comment by M-Davies** [comment]
@johnoliver It looks like `jenkinsBuildRoot` isn't getting set when I kick off a regenerator. See https://ci.adoptopenjdk.net/job/build-scripts/job/utils/job/pipeline_jobs_generator_jdk11u/71/console
```
[INFO] Running regeneration script...
[Pipeline] stage
[Pipeline] { (Check jdk11 pipeline status)
[Pipeline] echo
[ERROR] Failure on jenkins api connection or parsing.
Error: java.net.MalformedURLException: no protocol: null/api/json?tree=jobs[name]&pretty=true&depth1
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[INFO] Cleaning up...
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
java.lang.NullPointerException: Cannot get property 'jobs' on null object
	at org.codehaus.groovy.runtime.NullObject.getProperty(NullObject.java:60)
	at org.codehaus.groovy.runtime.InvokerHelper.getProperty(InvokerHelper.java:174)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.getProperty(ScriptBytecodeAdapter.java:456)
	at org.kohsuke.groovy.sandbox.impl.Checker$7.call(Checker.java:355)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:357)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.getProperty(SandboxInvoker.java:29)
	at com.cloudbees.groovy.cps.impl.PropertyAccessBlock.rawGet(PropertyAccessBlock.java:20)
	at Regeneration.regenerate(Script3.groovy:322)
	at ___cps.transform___(Native Method)
	at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.get(PropertyishBlock.java:74)
	at com.cloudbees.groovy.cps.LValueBlock$GetAdapter.receive(LValueBlock.java:30)
	at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.fixName(PropertyishBlock.java:66)
	at sun.reflect.GeneratedMethodAccessor186.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
	at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
	at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
	at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:185)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:400)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$400(CpsThreadGroup.java:96)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:312)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:276)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:67)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE
```

**Pull request merged**